proxy/envoy/config.go

```diff
@@ -674,17 +674,21 @@ func buildInboundListeners(mesh *proxyconfig.MeshConfig, sidecar proxy.Node,
 				model.SortRouteRules(rules)
 				for _, config := range rules {
 					rule := config.Spec.(*proxyconfig.RouteRule)
+					var route *HTTPRoute
 					if rule.WebsocketUpgrade {
-						websocketRoute := buildInboundWebsocketRoute(rule, cluster)
-
+						route = buildInboundWebsocketRoute(rule, cluster)
+					} else if config.ConfigMeta.Name != "" {
+						route = buildInboundRoute(config, rule, cluster)
+					}
+					if route != nil {
```

Comment:
Is this needed on inbound only? Why not use it in outbound as well?

Reply:
The outbound is handled in `buildHTTPRoute` (route.go).